### PR TITLE
Filter cases by internal id

### PR DESCRIPTION
**Pull request by seallard**
### Changed
- Replaced family function with get_case_by_internal_id

### Fixed
- Refactored filtering on internal id for cases

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg/store/api/find_business_data.py [hunk 1]
```diff
@@ -550,3 +551,12 @@ def is_case_external(self, case_id: str) -> bool:
...
         return all(sample.application_version.application.is_external for sample in case.samples)
+
+    def get_case_by_internal_id(self, internal_id: str) -> Family:
+        """Get case by internal id."""
+
+        return apply_case_filter(
+            filter_functions=[CaseFilter.FILTER_BY_INTERNAL_ID],
+            cases=self._get_query(table=Family),
+            internal_id=internal_id,
+        ).first()
```
Comment (henrikstranneheim) [inline cg/store/api/find_business_data.py:562]: This kind of code is starting to look very clean! Great work @seallard @Vince-janv @ChrOertlin @diitaz93

File: cg/store/api/delete.py [hunk 2]
```diff
@@ -27,7 +27,7 @@ def delete_relationships_sample(self, sample: Sample) -> None:
     def delete_cases_without_samples(self, case_ids: List[str]) -> None:
         """Delete any cases specified in case_ids without samples."""
         for case_id in case_ids:
-            case: Family = self.Family.query.filter(Family.internal_id == case_id).first()
+            case: Family = self.get_case_by_internal_id(internal_id=case_id)
             if case and not case.links:
                 case.delete()
         self.commit()
```
Comment (ChrOertlin) [inline cg/store/api/delete.py:33]:
```Python
  def delete_cases_without_samples(self, case_internal_ids: List[str]) -> None:
        """Delete any cases specified in case_ids without samples."""
        for case_internal_id in case_internal_ids:
            case: Family = self.get_case_by_internal_id(internal_id=case_internal_id)
            if case and not case.links:
                case.delete()
        self.commit()

```

Comment (ChrOertlin) [reply cg/store/api/delete.py:33]: open to suggestions

File: cg/store/api/status.py [hunk 3]
```diff
@@ -225,7 +225,7 @@ def cases(
 
     def set_case_action(self, action: Literal[CASE_ACTIONS], case_id: str) -> None:
         """Sets the action of provided cases to None or the given action."""
-        case_obj: Family = self.Family.query.filter(Family.internal_id == case_id).first()
+        case_obj: Family = self.get_case_by_internal_id(internal_id=case_id)
         case_obj.action = action
```
Comment (ChrOertlin) [inline cg/store/api/status.py:229]:
```suggestion
        case: Family = self.get_case_by_internal_id(internal_id=case_internal_id)
        case.action = action
```

File: cg/store/api/status.py [hunk 4]
```diff
@@ -225,7 +225,7 @@ def cases(
 
     def set_case_action(self, action: Literal[CASE_ACTIONS], case_id: str) -> None:
```
Comment (ChrOertlin) [inline cg/store/api/status.py:226]:
```suggestion
    def set_case_action(self, action: Literal[CASE_ACTIONS], case_internal_id: str) -> None:
```

**Review by ChrOertlin — APPROVED** [review]

File: cg/store/api/status.py [hunk 5]
```diff
@@ -225,7 +225,7 @@ def cases(
...
         """Sets the action of provided cases to None or the given action."""
-        case_obj: Family = self.Family.query.filter(Family.internal_id == case_id).first()
+        case_obj: Family = self.get_case_by_internal_id(internal_id=case_id)
```
Comment (henrikstranneheim) [inline cg/store/api/status.py:228]:
```suggestion
        case: Family = self.get_case_by_internal_id(internal_id=case_id)
```

**Review by henrikstranneheim — APPROVED** [review]
💯

File: cg/store/api/delete.py (hunk 2 above)
Comment (seallard) [reply cg/store/api/delete.py:33]: Updated the name to case_internal_ids

**Review by ChrOertlin — APPROVED** [review]
Focussed on that no attribute got changed, looks good!

**Review by seallard — COMMENTED** [review]
Went through the renaming, looks good.

**Pull request merged**